action/metadata-support.go

```diff
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"context"
+	"os"
+	"strings"
+
+	"github.com/google/go-github/v45/github"
+	"github.com/sethvargo/go-githubactions"
+	"golang.org/x/oauth2"
+)
+
+// getArtifacts queries github for artifact list, returning a 2 dimensional slice of
+// artifact names paired with their variant type.
+// org, repo, and workflowrunID are the github vars passed in to the API for the
+// workflow run query, see: https://docs.github.com/en/rest/actions/artifacts#list-workflow-run-artifacts
+func getArtifacts(org string, repo string, workflowRunID int64) [][]string {
+
+	// Auth to github API.
+	token := os.Getenv("CRT_GITHUB_TOKEN")
+	if token == "" {
+		githubactions.Errorf("missing env CRT_GITHUB_TOKEN")
+	}
+
+	ctx := context.Background()
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	ghClient := github.NewClient(tc)
+
+	// Query API for list of artifacts associated with specified build workflow of product.
+	artifacts, _, err := ghClient.Actions.ListWorkflowRunArtifacts(ctx, org, repo, workflowRunID, &github.ListOptions{})
```

Comment:
`ListWorkflowRunArtifacts` is paginated, and the default value is smaller than the number of artifacts some of our repositories produce.  We should increase the `PerPage` to the maximum of 100 in ListOptions{} and iterate over all the pages to collect the artifacts.

https://pkg.go.dev/github.com/google/go-github/v45/github#ListOptions
https://docs.github.com/en/rest/actions/artifacts#list-workflow-run-artifacts